api/agentpb/agent.proto

```diff
@@ -534,6 +534,25 @@ message JobProgress {
   }
 }
 
+// GetVersionRequest represents a request for version retrieving of different software.
+message GetVersionRequest {
```

Comment:
I would suggest to make it return all versions in one response instead of one per request, what do you think?

Reply:
To check the version of remote MongoDB we should pass DB creds. We couldn't find a better way, maybe you have some ideas?

Reply:
We can send list of messages. each message will have the same format that we use for `CheckConnectionRequest`
```
 // Service type.
  inventory.ServiceType type = 1;
  // DSN for the service. May contain connection (dial) timeout.
  string dsn = 2;
  // Timeout for the whole request.
  google.protobuf.Duration timeout = 3;
  // Contains files and their contents which can be used in DSN.
  TextFiles text_files = 4;
  // TLS certificate wont be verified.
  bool tls_skip_verify = 5;
```

And instead of checking for version right before restore we can do it periodically and store version in DB.
What do you think?

Reply:
we should guarantee the version of the db at the time of making a backup. That could be the issue in the case of periodical checking.

Reply:
In general, I prefer to have one request to fetch all versions instead of multiple requests.